src/app/static/src/app/store/password/actions.ts

```diff
@@ -2,27 +2,46 @@ import {ActionContext, ActionTree, Payload} from "vuex";
 import {PasswordState} from "./password";
 import {api} from "../../apiService";
 
-export type PasswordActionTypes = "ResetLinkRequested" | "RequestResetLinkError"
+export type PasswordActionTypes = "ResetLinkRequested" | "RequestResetLinkError" | "ResetPassword" | "ResetPasswordError"
 export interface PasswordActionPayload<T> extends Payload {
     type: PasswordActionTypes
     payload: T
 }
 
+export interface ResetPasswordActionParams {
+    token: string
+    password: string
+}
+
 export interface PasswordActions {
     requestResetLink: (store: ActionContext<PasswordState, PasswordState>, email: string) => void
+    resetPassword: (store: ActionContext<PasswordState, PasswordState>, payload: ResetPasswordActionParams) => void
 }
 
 export const actions: ActionTree<PasswordState, PasswordState> & PasswordActions = {
 
-    requestResetLink({commit}, email) {
+    requestResetLink({commit}: ActionContext<PasswordState, PasswordState>, email: string) {
         let formData = new FormData();
         formData.append('email', email);
         api.postAndReturn<null>("/password/request-reset-link/", formData)
             .then((payload) => {
-                commit<PasswordActionPayload<null>>({type: "ResetLinkRequested", payload});
+                commit<PasswordActionPayload<null>>({type: "ResetLinkRequested", payload: null});
             })
             .catch((error: Error) => {
                 commit<PasswordActionPayload<string>>({type: "RequestResetLinkError", payload: error.message});
             });
+    },
+
+    resetPassword({commit}: ActionContext<PasswordState, PasswordState>, payload: ResetPasswordActionParams) {
+        let formData = new FormData();
```

Comment:
Can we avoid using `FormData` unless we absolutely have to? Because it's a browser API it's a massive pain to test. If we're just passing simple data it's easiest just to stringify it, as I've done here: https://github.com/mrc-ide/hint/pull/29/files#diff-0f73780710f9ddc4b1ffe4acd2956a6dR22

Reply:
Fair enough for the reset link request, but I wouldn't have thought that was ideal for passwords? Gets encrypted over https, but hangs around in plain text on client (browser history) and server logs.